### DISCUSSION TITLE
In the Editor, the "matras" should not be considered as an additional character

### DISCUSSION THREAD

**Discussion by vinod303** [Sunbird-Knowlg/Community#77] [Feature Requests]
Team,
It has been observed that in the course editor , while writing a description & title (in the Hindi language), "MATRA" is getting counted as a character.  Ideally, this should NOT be counted as a character. 
Requesting the team to address this issue.

**Comment by vaibhavbhuva** [comment]
@vinod303 The character count calculation is happening based on the Unicode entered in the input box.  This can be configurable as part of the primary category form(Object category definition).
For example:
```
{
    code: 'name',
    ...
    validations: [
        {
            type: 'max',
            value: '120',
            message: 'Input is Exceeded'
        }
    ]
}
```

If you still feel this should NOT be counted as a character then please provide more details with examples.  We will check and respond back as soon as possible.

Thanks,
Vaibhav Bhuva

**Reply by kameshbhr** [reply]
From an end user point of view, a "character" means a "letter" (alphabet in case of English). For example the English word "Mathematics" has 11 characters. The Hindi word "गणित" has 3 characters (letters). If you count unicode characters for this Hindi word, it is 4 - because the letter "णि" has two unicode chars - one for the consonant and another for the vowel.

For an end user however, the word गणित has only 3 characters (letters). So when we show a character limit on the screen, we should only count the letters, not the unicode characters. For example, if the editor is showing the maximum character limit of the name as say, 4 and when user types in "गणित" the editor gives a message that the name is exceeding the character limit, the user wouldn't understand.

Hope this gives clarity about the problem.

**Reply by vaibhavbhuva** [reply]
@vinukumar-vs

**Reply by vinukumar-vs** [reply]
@vaibhavbhuva Take a look at this sample
https://jsbin.com/cisacaw/edit?html,js,console,output

Check can we integrate the same to get characters count.
@Sunbird-Knowlg/maintainers

**Reply by gouravmore** [reply]
We have checked the above and some more solutions to fix this character count issue. Here are the findings.
Google doc link:- https://docs.google.com/document/d/1uPSmjQo6jq3xHMsR9AsrsVUx0BpKZYwkhrHz9pmY9AA/edit?usp=sharing
